### ISSUE TITLE
Receive HID packets

### ISSUE THREAD

**Issue by raspberryvalley** [sandyjmacdonald/keybow2040-circuitpython#8]
Hi, I'm searching high and low for a solution to receive packets of data on the keybow. While sending in HID mode is available in examples and the Adafruit library, I'd appreciate an example to receive packets.

My scenario: any button on the keybow is in 'master' mode. I.e. the device it controls on my network is controlled from the pad.  Pad to device. I use it to control IoT devices. However, I would like to read events or statuses, say, by changing button color. Device on network to button on keybow.

Thanks for any tips. MQTT status reading and sending to keybow would be awesome

Thanks

**Comment by Gadgetoid** [comment]
I think the typical approach to this - at least if you're not a vendor writing your own driver - is to use USB serial. I'm not sure how you'd go about doing this on CircuitPython alongside HID but I'd wager it's possible. Then you can use a Python script or similar to push serial status updates back to the device.

**Comment by raspberryvalley** [comment]
Thanks @Gadgetoid , appreciate the tip. 

I have to sort it out and will share once I do. Thanks

**Comment by mosch** [comment]
@raspberryvalley did u come up with something here?

**Comment by mschwld** [comment]
@mosch I solved a scenario similar to the one @raspberryvalley described using USB serial communications utilizing this https://circuitpython.readthedocs.io/en/latest/shared-bindings/usb_cdc/index.html. The Keybow2040 sends each number of a pressed key which will trigger certain actions on the host (which can also rerspect layer definitions). In return it receives updates in the form of `[keynumber],[colorcode]` to control the LEDs. I uploaded the code here: https://gist.github.com/mschwld/7e0e16fbc8a7dc6f24b5d31c1f946329.
